pandas/plotting/_core.py

```diff
@@ -1746,69 +1746,66 @@ def _find_backend(backend: str):
     types.ModuleType
         The imported backend.
     """
-    import pkg_resources  # Delay import for performance.
-
-    for entry_point in pkg_resources.iter_entry_points("pandas_plotting_backends"):
-        if entry_point.name == "matplotlib":
-            # matplotlib is an optional dependency. When
-            # missing, this would raise.
-            continue
-        _backends[entry_point.name] = entry_point.load()
-
-    try:
-        return _backends[backend]
-    except KeyError:
-        # Fall back to unregistered, module name approach.
+    if backend == "matplotlib":
+        # Because matplotlib is an optional dependency and first-party backend,
+        # we need to attempt an import here to raise an ImportError if needed.
         try:
-            module = importlib.import_module(backend)
+            import pandas.plotting._matplotlib as module
         except ImportError:
-            # We re-raise later on.
-            pass
-        else:
-            if hasattr(module, "plot"):
-                # Validate that the interface is implemented when the option
-                # is set, rather than at plot time.
-                _backends[backend] = module
-                return module
+            raise ImportError(
+                "matplotlib is required for plotting when the "
+                'default backend "matplotlib" is selected.'
+            ) from None
+
+    else:
+        module = None
+        # Delay import for performance.
+        # TODO: replace with `importlib.metadata` when python_requires >= 3.8.
+        from pkg_resources import iter_entry_points
```

Comment:
import this at the top

Reply:
I trusted the comment and left it there, I timed it at ~100ms on my machine, do you think it's ok then to import at the top?

Reply:
this is the std library

Reply:
It's not, it's part of `setuptools` and as said [in their docs](https://setuptools.readthedocs.io/en/latest/pkg_resources.html):

> Use of pkg_resources is discouraged in favor of importlib.resources, importlib.metadata, and their backports (resources, metadata). Please consider using those libraries instead of pkg_resources.

because of its inefficiency (see pypa/pip#7413). Since python 3.8 only have all (or almost) its functionalities been included in `importlib`, which is part of the standard library.